src/persist-client/src/internal/trace.rs

```diff
@@ -529,9 +571,8 @@ impl<T: Timestamp + Lattice> Trace<T> {
     ) -> Vec<FueledMergeReq<T>> {
         // Returns true if b0 covers b1, false otherwise.
         fn covers<T: PartialOrder>(b0: &FueledMergeReq<T>, b1: &FueledMergeReq<T>) -> bool {
-            PartialOrder::less_equal(b0.desc.lower(), b1.desc.lower())
-                && PartialOrder::less_equal(b1.desc.upper(), b0.desc.upper())
-                && b0.desc.since() == b1.desc.since()
+            // TODO: can we relax or remove this since check?
+            b0.id.covers(b1.id) && b0.desc.since() == b1.desc.since()
```

Comment:
This is probably totally fine by I don't really understand the change here. Why did we go from comparing the 'lower' and 'upper' on the description to the 'id'?

Reply:
One way to think of the `SpineId` is that it's a bit redundant with the lower and upper in the desc - they both mark intervals of some space, and the upper end of each batch is supposed to match the lower end of the other - but the spine id is a bit easier to deal with. (Totally ordered, monotonically increasing, etc.)

The purpose of this code is to avoid passing on merge requests that are just subsets of other, larger merge requests. In this case either bound is fine to use, so might as well use the simpler one / the one that reflects the tree structure more directly. (Or: might as well leave it alone for now, if you prefer keeping this diff small!)